LTHPasscodeViewController/LTHPasscodeViewController.m

```diff
@@ -732,79 +737,56 @@ - (void)updateViewConstraints {
-                                     relatedBy: NSLayoutRelationEqual
-                                        toItem: _animatingView
-                                     attribute: NSLayoutAttributeCenterX
-                                    multiplier: 1.0f
-                                      constant: _horizontalGap - 2.0f];
-        NSLayoutConstraint *firstDigitY =
-        [NSLayoutConstraint constraintWithItem: _firstDigitTextField
-                                     attribute: NSLayoutAttributeCenterY
-                                     relatedBy: NSLayoutRelationEqual
-                                        toItem: _enterPasscodeLabel
-                                     attribute: NSLayoutAttributeBottom
-                                    multiplier: 1.0f
-                                      constant: _verticalGap];
-        NSLayoutConstraint *secondDigitY =
-        [NSLayoutConstraint constraintWithItem: _secondDigitTextField
-                                     attribute: NSLayoutAttributeCenterY
-                                     relatedBy: NSLayoutRelationEqual
-                                        toItem: _enterPasscodeLabel
-                                     attribute: NSLayoutAttributeBottom
-                                    multiplier: 1.0f
-                                      constant: _verticalGap];
-        NSLayoutConstraint *thirdDigitY =
-        [NSLayoutConstraint constraintWithItem: _thirdDigitTextField
-                                     attribute: NSLayoutAttributeCenterY
-                                     relatedBy: NSLayoutRelationEqual
-                                        toItem: _enterPasscodeLabel
-                                     attribute: NSLayoutAttributeBottom
-                                    multiplier: 1.0f
-                                      constant: _verticalGap];
-        NSLayoutConstraint *fourthDigitY =
-        [NSLayoutConstraint constraintWithItem: _fourthDigitTextField
-                                     attribute: NSLayoutAttributeCenterY
-                                     relatedBy: NSLayoutRelationEqual
-                                        toItem: _enterPasscodeLabel
-                                     attribute: NSLayoutAttributeBottom
-                                    multiplier: 1.0f
-                                      constant: _verticalGap];
-        [self.view addConstraint:firstDigitX];
-        [self.view addConstraint:secondDigitX];
-        [self.view addConstraint:thirdDigitX];
-        [self.view addConstraint:fourthDigitX];
-        [self.view addConstraint:firstDigitY];
-        [self.view addConstraint:secondDigitY];
-        [self.view addConstraint:thirdDigitY];
-        [self.view addConstraint:fourthDigitY];
+        
+        float delta = 10.0f;
+        
+        for (int i = 0; i < _digitsCount; ++i) {
+            
+            float constant = _digitsCount % 2 == 0
```

Comment:
I don't have access to the code, but I think they were like that before. Since this is a bit more complex, I think doing it starting from the leading would be easier to reason with. `digits * (fieldWidth + padding) - padding` easily gives you the width of all elements + the paddings, then the difference up to screen width just has to be halved. 

Or, even better: we could add all fields in a container, and all there's left to do then, is to center that one horizontally - this means its constraints don't have to be updated anymore or rotation, either.